src/main/scala/org/ergoplatform/nodeView/wallet/ErgoWalletActor.scala

```diff
@@ -142,44 +146,75 @@ class ErgoWalletActor(settings: ErgoSettings) extends Actor with ScorexLogging {
       height = heightTo
   }
 
-  protected def generateTransactionWithOutputs(payTo: Seq[ErgoBoxCandidate]): Try[ErgoTransaction] = Try {
-    require(prover.dlogPubkeys.nonEmpty, "No public keys in the prover to extract change address from")
-    require(payTo.forall(_.value > 0), "Non-positive Ergo value")
-    require(payTo.forall(_.additionalTokens.forall(_._2 > 0)), "Non-positive asset value")
+  private def requestsToBoxCandidates(requests: Seq[TransactionRequest]): Try[Seq[ErgoBoxCandidate]] = Try {
+    requests.map {
+      case PaymentRequest(address, value, assets, registers) =>
+        new ErgoBoxCandidate(value, address.script, assets.getOrElse(Seq.empty), registers.getOrElse(Map.empty))
+      case AssetIssueRequest(address, amount, name, description, decimals) =>
+        val firstInput = inputsFor(
+          requests
+            .foldLeft(Seq[PaymentRequest]()) {
+              case (acc, pr: PaymentRequest) => acc :+ pr
+              case (acc, _) => acc
+            }
+            .map(_.value)
+            .sum
+        ).head
+        val assetId = Digest32 !@@ firstInput.id
+        val nonMandatoryRegisters = scala.Predef.Map(
+          R4 -> StringConstant(name),
+          R5 -> StringConstant(description),
+          R6 -> IntConstant(decimals)
+        )
+        new ErgoBoxCandidate(0L, address.script, Seq(assetId -> amount), nonMandatoryRegisters)
+      case other => throw new Exception(s"Unknown TransactionRequest type: $other")
+    }
+  }
 
-    val targetBalance = payTo.map(_.value).sum
+  protected def generateTransactionWithOutputs(requests: Seq[TransactionRequest]): Try[ErgoTransaction] =
+    requestsToBoxCandidates(requests).map { payTo =>
+      require(prover.dlogPubkeys.nonEmpty, "No public keys in the prover to extract change address from")
+      require(payTo.forall(_.value >= 0), "Non-positive Ergo value")
+      require(payTo.forall(_.additionalTokens.forall(_._2 >= 0)), "Non-positive asset value")
```

Comment:
I think we should also require at most 1 AssetIssueRequest

Reply:
done